### PR TITLE
Check Python formatting in pre-commit hook & re-format py files

### PR DESCRIPTION
**Pull request by letmaik**
This adds [black](https://black.readthedocs.io) as Python code formatter to the checks in the pre-commit hook and CI.

cc @wintersteiger

### CODE REVIEW

**Comment by letmaik** [comment]
@wintersteiger This re-formats sgx-lkl-cfg as well which you're modifying slightly in your PR. To avoid merge conflicts simply run `black tools/sgx-lkl-cfg` before merging. You can install black with `python3 -m pip install black`.

**Comment by wintersteiger** [comment]
I black-formatted sgx-lkl-cfg, but I usually rebase my PR branches, so I'll have to keep re-formatting it I guess.

**Comment by SeanTAllen** [comment]
@wintersteiger are you familiar with `rerere`? https://git-scm.com/docs/git-rerere

**Comment by wintersteiger** [comment]
@SeanTAllen I wasn't, thanks for the pointer, that looks useful!

**Comment by letmaik** [comment]
Could *someone* review this?

File: scripts/check-ci [hunk 1]
```diff
@@ -13,10 +13,15 @@ exit_() {
     exit 1
 }
 
-## Check whether source code has been formatted:
-if ! ./scripts/format-code --quiet --whatif; then
-    echo "code not formatted correctly, IGNORING FOR NOW!"
-    #exit_ "CI failed: please run ./scripts/format-code"
+## Check whether C/C++ source code has been formatted:
```
Comment (wintersteiger) [inline scripts/check-ci:16]: Good idea to keep it a warning for now. Am I right that `format-c-code` checks/formats only the files in a (staged?) commit?

File: samples/ml/pytorch/app/sample.py [hunk 2]
```diff
@@ -15,17 +15,15 @@
...
 )
 
 # The nn package also contains definitions of popular loss functions; in this
 # case we will use Mean Squared Error (MSE) as our loss function.
-loss_fn = torch.nn.MSELoss(reduction='sum')
+loss_fn = torch.nn.MSELoss(reduction="sum")
 
 learning_rate = 1e-4
-print('Training...')
+print("Training...")
```
Comment (wintersteiger) [inline samples/ml/pytorch/app/sample.py:26]: Is the latest fashion trend in the Python world to use `"` instead of `'`?

**Review by wintersteiger — APPROVED** [review]

File: scripts/check-ci (hunk 1 above)
Comment (letmaik) [reply scripts/check-ci:16]: This PR doesn't enable C code formatting. I haven't looked at the details of that yet.

Comment (wintersteiger) [reply scripts/check-ci:16]: Sure, sounds good. It just looked like you're calling the C formatting script here.

**Pull request merged**